src/api/features.api.js

```diff
@@ -17,72 +18,220 @@ export class Feature extends EventEmitter {
      */
-    constructor(id, coordinate, title, description, isEditable = false) {
+    constructor(id, coordinates, title, description, isEditable = false) {
         super()
         this._id = id
-        this._coordinate = [...coordinate]
+        // using the setter for coordinate (see below)
+        this.coordinates = coordinates
         this._title = title
         this._description = description
         this._isEditable = !!isEditable
+        this._isDragged = false
     }
 
-    emitChangeEvent() {
+    /**
+     * Emits a change event and, if changeType is defined, a change:changeType event
+     *
+     * @param {String} changeType So that the change event is more specific, for instance if
+     *   changeType is equal to 'title' a 'change' event and a 'change:title' event will be fired
+     */
+    emitChangeEvent(changeType = null) {
         this.emit('change', this)
+        if (changeType) {
+            this.emit(`change:${changeType}`, this)
+        }
     }
 
     get id() {
         return this._id
     }
     // ID is immutable, no setter
 
-    get coordinate() {
-        return this._coordinate
+    get coordinates() {
+        return this._coordinates
     }
-    set coordinate(newCoordinate) {
-        if (Array.isArray(newCoordinate) && newCoordinate.length === 2) {
-            this._coordinate = newCoordinate
+    set coordinates(newCoordinates) {
+        if (Array.isArray(newCoordinates)) {
+            // checking if we have received a unique coordinate
+            if (
+                newCoordinates.length === 2 &&
+                newCoordinates.filter((coord) => Array.isArray(coord)).length === 0
+            ) {
+                // as we want an array of coordinates, we wrap the unique coordinate in an array
+                this._coordinates = [newCoordinates]
+            } else {
+                this._coordinates = newCoordinates
+            }
             this.emitChangeEvent()
         }
     }
+    get lastCoordinate() {
+        return this._coordinates[this._coordinates.length - 1]
+    }
 
     get title() {
         return this._title
     }
     set title(newTitle) {
         this._title = newTitle
-        this.emitChangeEvent()
+        this.emitChangeEvent('title')
     }
 
     get description() {
         return this._description
     }
     set description(newDescription) {
         this._description = newDescription
-        this.emitChangeEvent()
+        this.emitChangeEvent('description')
     }
 
     get isEditable() {
         return this._isEditable
     }
     // isEditable is immutable, no setter
+
+    /**
+     * Tells if the feature is currently being dragged (and later dropped) by the user
+     *
+     * @returns {boolean}
+     */
+    get isDragged() {
+        return this._isDragged
+    }
+    set isDragged(flag) {
+        this._isDragged = flag
+        this.emitChangeEvent('isDragged')
```

Comment:
Shouldn't this only be emitted if `flag === true`?

Reply:
mmh, yes and no, it means the value of `isDragged` has changed (an event called `change:isDragged` will be fired), maybe the naming is not good enough.

Reply:
Ok. I think I get it. So, we' use it like this?
`.on('isDragged', (feature) => if (feature.isDragged) { ... })`

Reply:
yep, that's the idea

Reply:
:+1: